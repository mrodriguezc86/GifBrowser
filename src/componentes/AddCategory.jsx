import PropTypes from "prop-types";
import { useState } from "react";


export const AddCategory = ({onSubmitCategory}) => {
    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length<= 1) return;
        onSubmitCategory(inputValue)
        //setCategory(category => [inputValue, ...category])
        setInputValue('');
    }

    return(
        <form onSubmit={(event) => onSubmit(event)} aria-label="form">
            <input
                type='text'
                placeholder="Find gifs"
                value={inputValue}
                onChange={onInputChange}
            ></input>
        </form>
    )
}

AddCategory.propTypes = {
    onSubmitCategory: PropTypes.func.isRequired,
}