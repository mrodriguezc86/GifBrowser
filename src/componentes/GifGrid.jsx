import PropTypes from 'prop-types'
import { getGifs } from '../helpers/getGifs';
import { useEffect, useState } from 'react';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category}) => {

  const {images, isLoading} = useFetchGifs( category )
   

  return (
    <div>
      <h3>{category}</h3>

      {isLoading && (<h2>Loading...</h2>)}

      <div className='card-grid'>
        {images.map((image)=>{
          return <GifItem key={image.id} {...image}></GifItem>
        })}

      </div>
      
    
    </div>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}

