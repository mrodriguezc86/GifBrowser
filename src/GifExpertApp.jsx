import { useState } from "react"
import {AddCategory, GifGrid} from './componentes'

export const GifExpertApp = () => {

    const [category, setCategory] = useState([])

    const onSubmitCategory = (inputValue) => {
        if(category.includes(inputValue))return;
        setCategory([inputValue, ...category])
    }

  return (
    <>
        <h1>Gif Browser</h1>
        <AddCategory onSubmitCategory={onSubmitCategory}/>
        
       
            {
                category.map(category => (
                    <GifGrid key={category} category={category}></GifGrid>
                ))
            }
           
        
    </>
  )
}
