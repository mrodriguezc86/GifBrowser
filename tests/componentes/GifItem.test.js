import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/componentes";

describe('Test component <GetItem/>', () => { 

    const title = 'hello'
    const url = 'https://www.hello.com/'
    
    test('should match the snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
     })

     test('should show url image and alt text', () => { 
        render(<GifItem title={title} url={url}/>)
        //screen.debug();
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
      })

      test('should show Title', () => { 
        render(<GifItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();
       })
 })