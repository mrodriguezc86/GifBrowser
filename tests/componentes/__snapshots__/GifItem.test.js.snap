// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <GetItem/> should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="hello"
      src="https://www.hello.com/"
    />
    <p>
      hello
    </p>
  </div>
</div>
`;
